osuosl/timesync-frontend-flask
# Jerryp/fix login button when logged in

Fixes #173 #152 #171 #151 #177 #179 #156 #181 #182 #174 
#179 and #156 is the same problem
## Changes in this PR.

- [X] Take out login button on side menu when user is logged in
- [X] Added break after submit buttons
- [X] Added pagination with materialize integrated
- [X] Fixed sidebar Icon glitch
- [X] Fixed makeSortable function glitch
- [X] Fixed site admin project viewability in create_time, view_time page
- [X] Fixed clear form glitches
- [X] Added format example for unit measurement for "Durarion" in form

## Testing this PR.
1. docker-compose up
2. login

To test login button:
Check login button not present when logged in.

To test pagination:
Go to all the view_* pages and test pagination see if it works.

To test sidebar icon:
Shrink the width of page until the icon appears on top bar. Click to test.

To test makeSortable:
Click on a tab where it is not sortable.

To test Admin project viewability:
Login as site admin and see if all project slugs show up in the form for create_time, view_time page.
Login as non site admin and see if only user specific project slugs show up.

To test clearForm.js:
Click on clear button and see if all the options are unclicked.

To test object.css is not linked:
open view_activities.html, view_times.html, view_users.html

To test the duration format example:
open create_time.html, to see if there are any format examples for duration.

### Expected Output.

1. Login button not present when logged in
2. Pagination works and does not have glitches when clicking on it. works with makesort() as well.
3. Clicking sidebar icon will open the side bar.
4. The table is not modified when tab is clicked.
5. All the project slugs show up in the form for create_time, view_time page only when user is site admin, if user is not, then only user specific project slug show up.
6. All the buttons are unclicked, and for view_user and view_activities, the table is reloaded.
7. All the html templates do not include object.css.
8. The create_time.html has format example for duration.
